WalletWasabi.Fluent/ViewModels/Wallets/Actions/Receive/ReceiveAddressViewModel.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Reactive.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Windows.Input;
+using Avalonia;
+using Gma.QrCodeNet.Encoding;
+using NBitcoin;
+using ReactiveUI;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Fluent.ViewModels.Navigation;
+using WalletWasabi.Hwi;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.Fluent.ViewModels.Wallets.Actions.Receive
+{
+	public partial class ReceiveAddressViewModel : RoutableViewModel
+	{
+		[AutoNotify] private bool _animationTrigger;
+		[AutoNotify] private string _actionText;
+		[AutoNotify] private string? _gapLimitMessage;
+
+		public ReceiveAddressViewModel(HdPubKey model, Network network, HDFingerprint? masterFingerprint, bool isHardwareWallet, string? minGapLimitMessage)
+		{
+			Title = "Address";
+			Address = model.GetP2wpkhAddress(network).ToString();
+			Reference = model.Label;
+			IsHardwareWallet = isHardwareWallet;
+			_actionText = "";
+			_gapLimitMessage = minGapLimitMessage;
+
+			GenerateQrCode();
+
+			CopyAddressCommand = ReactiveCommand.CreateFromTask(async () =>
+			{
+				ActionText = "Copied";
```

Comment:
I wonder if we should bring back notifications for this kind of thing?

Reply:
Solves all the style trigger hacks and a consistent way to display these types of things in the ui

Reply:
Agree with that, I wondered about the same when I worked on other PRs.